### ISSUE TITLE
License

### ISSUE THREAD

**Issue by m1cm1c** [Beagle-PSE/Beagle#754]
We already talked about choosing a license and we should do so. I suggest using either `GNU GPL v2.0` or the `MIT License`. As we all participated in this project, all of us need to agree to a license for Beagle to be under. Please proclaim your thoughts about either or both licenses here or suggest a different one.

**Comment by jGleitz** [comment]
>  I suggest using either [`GNU GPL v2.0`](http://choosealicense.com/licenses/gpl-2.0/) or the [`MIT`](http://choosealicense.com/licenses/mit/) License

I fully agree to use either `GPL` or [`MIT`](http://choosealicense.com/licenses/mit/). However, why not use [`GPL v3`](http://choosealicense.com/licenses/gpl-3.0/) (don’t say because Linus doesn’t like it)?

Generally, I prefer the `GPL` over the `MIT`, because it prevents that Open Source is used to build closed source products. However, we have to check with the SDQ whether that’s okay for them.

Palladio is licensed under [`EPL`](http://choosealicense.com/licenses/epl-1.0/), which is, as far as I can see, similar to the [`LPGL`](http://choosealicense.com/licenses/lgpl-3.0/)

**Comment by m1cm1c** [comment]
I disagree with Linus on this issue and am with Stallman. Other than Linus I'm against tivoization. Basically all I dislike about version 3 is that it tries to introduce protection against tivoization except it doesn't because you can sublicense anything licensed under version 3 forbidding tivoization under license 3 allowing tivoization and then get a copy of the fork allowing tivoization and do whatever tivoization you like with it. I think this is stupid.

**Comment by m1cm1c** [comment]
So to clarify: I'm not against version 3 and am fine with using it. I just think that it's stupid.

**Comment by jGleitz** [comment]
>  I'm not against version 3

Okay. So if we use GPL, let’s use GPL v3, as it is the recommended version and [contains further improvements](http://gplv3.fsf.org/rms-why.html).

To the things you said: I’d be pretty surprised if that’s true. Do you have sources for it? Because the GPL states “If you include code under this license in a larger program, the larger program must be under this license too”. So if somebody takes a GPLv3 project and introduces changes, he has to publish it under GPLv3. So I can take the code, remove the parts I don’t like about it (like DRM) but still keep the parts that somebody added to his fork. That’s prevention of tivoization, isn’t it?

**Comment by m1cm1c** [comment]
No, tivoization is about making the users unable to control software in practice even though from a legal perspective they have to right to control it.

GPL v3 doesn't allow tivoization in general. It only allows it for certain cases. The problem is that GPL v3 is compatible to itself even if one of the instances permits tivoization and the other one doesn't. Then source code can flow from a project forbidding tivoization to one allowing tivoization without the creators of the one allowing tivoization having to declare a separate license for the parts they took from a tivoization-forbidding project. Consequently, this means that (through a project allowing tivoization) source code of a project forbidding tivoization can be used for tivoization.

That's actually one of Linus' arguments against version 3: https://youtu.be/PaKIZ7gJlRU?t=3m19s

**Comment by jGleitz** [comment]
I watched that talk and at that time, it left me more confused than it explained. I just don’t get it. I don’t see why the thing you’re stating should be possible, as I think they’re explicitly forbidden. But let’s stop this discussion as it’s off-topic.

What do the others think about the licenses?

**Comment by viridispirus** [comment]
I think GPL v3 fit for our project.

**Comment by jGleitz** [comment]
The SDQ does not put any restrictions on which license we choose. However, they pointed out that the GPL could cause problems with Eclipse. This needs further clarification.

Furthermore, our supervisors pointed out that the GPL could be a reason for companies not using the software because they are unable to sell binaries containing it without publishing the source. As a consequence, the project could be less useful. We need to consider this and decide whether that’s what we want (because we think the GPL is the “right” way).

**Comment by jGleitz** [comment]
The SDQ suggests the [`EPL`](http://choosealicense.com/licenses/epl-1.0/), which seems to be the default for Eclipse Plugins.
